e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,141 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	"cosmossdk.io/math"
+	fiattokenfactorytypes "github.com/circlefin/noble-fiattokenfactory/x/fiattokenfactory/types"
+	"github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFiatTFUpdateOwner(t *testing.T) {
```

Comment:
What's the rationale for one big test with multiple cases in it, rather than smaller tests which each assert one thing? I noticed this particularly in the below test where you needed to track what state the system is currently in based on what's already happened in order to determine what result you expect.

Reply:
Mainly test efficiency. These are end to end so the chain is started from genesis and a validator is created. In some cases multiple chains are created with multiple validators along with relayers and ibc-paths. 

To me, that felt like a lot of overhead to test these specifics. I do however think there is a case to have more broken down testcases when it comes to unit testing.